src/kartograf/atom_mapper.py

```diff
@@ -855,9 +854,76 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    @staticmethod
+    def _split_protein_components_molecules(protein: ProteinComponent) -> list[ProteinComponent]:
+        """
+        Aims at splitting a protein component into different protein components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a list of ``ProteinComponent`` objects with the name attribute indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = protein.to_rdkit()
+        index_tuples = []
+        fragments = GetMolFrags(rdmol, asMols=True, sanitizeFrags=True, fragsMolAtomMapping=index_tuples)
+        components = [ProteinComponent(fragment, name=f"frag_{index_tuple[0]}") for fragment, index_tuple in zip(fragments, index_tuples)]
+
+        return components
+
+    def suggest_protein_mapping(self, protein_a: ProteinComponent, protein_b: ProteinComponent) -> LigandAtomMapping:
```

Comment:
Looking at this code, very little here is specific to ProteinComponent.

I.e. you could see a similar case where you have an SMC with multiple molecules (I think I remember this is a thing you can actually do - I remember someone in industry benchmarks stuffing disconnected molecules in the same SDF molecule).

It may make sense here to have this be the main driver for mapping going forward? I.e. by default you always try to pick up all the disconnected molecules and map them. So that we don't end up converting from ProteinComponent to SMC or otherwise, maybe we can check for the incoming component object type and reconstruct the outgoing component with the same object type?

What do folks think?

Reply:
Yeah I think having a single method which is agnostic to the component would be helpful and streamline this more I'll make that change. One possible issue is if the user accidentally supplies a disconnected SMC this will no longer raise an error, maybe this is more of a gufe issue and we should have SMC check if its connected or not when its created?

Reply:
That might be a discussion to have - I can think of a few cases where you might want to have disconnected molecules being treated together. It might need to be a protocol level thing.

Reply:
Yeah I think it would make sense for the protocol to dictate if it can handle that or not!